nomad/fsm_test.go

```diff
@@ -999,8 +1018,23 @@ func TestFSM_SnapshotRestore_AddMissingSummary(t *testing.T) {
 		CreateIndex: latestIndex,
 		ModifyIndex: latestIndex,
 	}
-
 	if !reflect.DeepEqual(&expected, out1) {
 		t.Fatalf("expected: %#v, actual: %#v", &expected, out1)
 	}
+
+	out2, _ := state2.JobSummaryByID(job2.ID)
+	expected = structs.JobSummary{
+		JobID: job2.ID,
+		Summary: map[string]structs.TaskGroupSummary{
+			"web": structs.TaskGroupSummary{
+				Queued:   10,
+				Starting: 0,
+			},
+		},
+		CreateIndex: latestIndex,
+		ModifyIndex: latestIndex,
+	}
+	if !reflect.DeepEqual(&expected, out2) {
+		t.Fatalf("expected: %#v, actual: %#v", &expected, out2)
+	}
```

Comment:
Add another job that has some allocs already such that there will be a starting, running, queued, complete, failed and lost count after reconcile